### PR TITLE
JP-1403 Fix handling of the boolean-like environmental variables PASS_INVALID_VALUES and STRICT_VALIDATION

### LINKED ISSUE #4777: Environmental variables should not simply override class init arguments
Issue [JP-1403](https://jira.stsci.edu/browse/JP-1403) was created by Jonathan Eisenhamer:

In `jwst.datamodels.model_base.DataModel`, two environmentals, if set, override the following class arguments:
 * PASS_INVALID_VALUES overrides pass_invalid_values
 * STRICT_VALIDATION overrides strict_validation

In general, environmentals should not blindly override function/class arguments. Environmentals should only be used if the arguments have not been explicitly set. If an argument has been explicitly set, the environmental should not be used.

### PR DESCRIPTION
**Pull request by stscieisenhamer**
Resolves #4777 
Resolves [JP-1403](https://jira.stsci.edu/browse/JP-1403)

### CODE REVIEW

File: jwst/datamodels/model_base.py [hunk 1]
```diff
@@ -306,14 +313,31 @@ def close(self):
             if fd is not None:
                 fd.close()
 
-    def get_envar(self, name, value):
+    def get_envar_as_boolean(self, name, default=False):
```
Comment (jdavies-st) [inline jwst/datamodels/model_base.py:316]: This should probably be a private method `_get_envar_as_boolean`.  I don't think we want it as part of the pubic API of the class.

File: jwst/datamodels/tests/test_models.py [hunk 2]
```diff
@@ -702,3 +703,75 @@ def test_abvega_offset_model():
...
+        ('y', True, True),
+        ('Y', False, True),
+        ('Y', True, True),
+        ('yes', False, True),
+        ('yes', True, True),
+        ('Yes', False, True),
+        ('Yes', True, True),
+        ('YES', False, True),
+        ('YES', True, True),
+    ]
```
Comment (jdavies-st) [inline jwst/datamodels/tests/test_models.py:763]: Did you get all permutations?  😂 👍

File: jwst/datamodels/tests/test_models.py [hunk 3]
```diff
@@ -702,3 +703,75 @@ def test_abvega_offset_model():
...
+
+@pytest.fixture(scope='module')
+def empty_model():
+    """Create an empy model"""
+    return DataModel()
+
+
+@pytest.fixture(scope='module')
+def jail_environ():
+    """Lock cheanges to the environment"""
```
Comment (jwhite3) [inline jwst/datamodels/tests/test_models.py:715]:
```suggestion
    """Lock changes to the environment"""
```

File: jwst/datamodels/tests/test_models.py (hunk 2 above)
Comment (stscieisenhamer) [reply jwst/datamodels/tests/test_models.py:763]: No, but there was actually good reason for these. One would be surprised...

The frustration level for this bit-o-code was higher than it really should have been.

**Comment by stscieisenhamer** [comment]
Not sure why the jenkins build is hung; it reports done and good. Merging.

**Pull request merged**